stats/cloud/collector.go

```diff
@@ -231,6 +238,8 @@ func (c *Collector) Run(ctx context.Context) {
 		case <-ctx.Done():
 			c.pushMetrics()
 			return
+		case <-c.stopCh:
+			return
```

Comment:
Highly unlikely but selects are not ordered and it is possible that the `pushTicker` has already ticked while we were pushing the previous metrics and even though the `stopCh` is closed, to make a pushMetrics. This is not a big deal in of itself but you are supposed to get that error again and then close the channel again which will definitely panic ;). (This could also happen because the ctx was `Done` ) 

I propose that you make another select above this one with just stopCh and a default clause. This way it's always checked and there is no way to change it outside of the `pushMetrics` so it will be fine

Reply:
You can probably add a test but it will be too indeterministic IMO

Reply:
Nice catch, fixed.